src/prefect/agent/kubernetes/agent.py

```diff
@@ -103,9 +105,177 @@ def __init__(
+                        metrics = self.api_client.call_api(
+                            f"/apis/metrics.k8s.io/v1beta1/namespaces/{self.namespace}/pods/{pod_name}",
+                            "GET",
+                            auth_settings=["BearerToken"],
+                            response_type="json",
+                            _preload_content=False,
+                        )
+                        response = json.loads(metrics[0].data.decode("utf-8"))
+
+                        for container in response["containers"]:
+                            name = container.get("name")
+                            cpu = container.get("usage").get("cpu")
+                            memory = container.get("usage").get("memory")
+                            self.logger.debug(
+                                f"Pod {pod_name} {name} container using {cpu} CPU and {memory} memory"
+                            )
+                    except self.k8s_client.rest.ApiException as exc:
+                        if exc.status != 404:
+                            self.logger.error(
+                                f"{exc.status} error attempting to read metrics of pod {pod_name}"
+                            )
+
+                    if not delete_job and pod_status.container_statuses:
+                        for container_status in pod_status.container_statuses:
+                            waiting = container_status.state.waiting
+                            if waiting and (
+                                waiting.reason == "ErrImagePull"
+                                or waiting.reason == "ImagePullBackOff"
+                            ):
+                                self.logger.debug(
+                                    f"Failing flow run {job['flow_run_id']} due to pod {waiting.reason}"
+                                )
+                                self.client.set_flow_run_state(
+                                    flow_run_id=job["flow_run_id"],
+                                    state=Failed(
+                                        message="Kubernetes Error: {}".format(
+                                            container_status.state.waiting.message
+                                        )
+                                    ),
+                                )
+
+                                delete_job = True
+                                break
+
+            if delete_job or job_status.failed or job_status.succeeded:
+                self.logger.debug(f"Deleting job {job['job_name']}")
+                try:
+                    self.batch_client.delete_namespaced_job(
+                        name=job["job_name"],
+                        namespace=self.namespace,
+                        body=self.k8s_client.V1DeleteOptions(
+                            propagation_policy="Foreground"
+                        ),
+                    )
+                except self.k8s_client.rest.ApiException as exc:
```

Comment:
Perhaps 404s should be exempt from logging, as they'd indicate the job was already deleted.